internal/wireguard/wg.go

```diff
@@ -141,6 +141,11 @@ func C25519pair() (string, string) {
 type WireGuardStates map[string]*wg.WireGuardState
 
 func GetWireGuardState() (WireGuardStates, error) {
+	// Re-read the config to make sure we are working with the latest on-disk version.
+	// Otherwise we might write stale data that was there when the agent was started
+	// (like deleted token).
+	flyctl.InitConfig()
```

Comment:
flyctl.InitConfig does a lot and it would even saves the config again.

https://github.com/superfly/flyctl/blob/d614f1ad9bd5ca729dc2de186cbf0f326d206819/flyctl/flyctl.go#L90-L114

What is the specific bug you'd like to fix? Is it possible to do this reload before pruning, instead of all Getter calls?

Reply:
> flyctl.InitConfig does a lot
I think I can do just `viper.ReadInConfig()`

>  and it would even save the config again.
Only if we are migrating from an old config format, no?

> What is the specific bug you'd like to fix?
Agent is long-running and right now operates on the cached copy of the config. Any changes made to the config get overwritten by it (like token, for example).

> Is it possible to do this reload before pruning, instead of all getter calls?
`GetWireGuardState() is also called from `validateTunnelsUnlocked` which terminates tunnels for invalid peers. So it needs up-to-date peer info as well.